### PR TITLE
View with ajax

### PR DESCRIPTION
**Pull request by gussiciliano**
### Sumary
- create a new book suggestion in a view usgin ajax for request
- list all book suggestions 

### Trello Card
https://trello.com/c/KFA9najK/28-crear-vista-para-crear-book-suggestions

### CODE REVIEW

File: app/assets/stylesheets/landing.scss [hunk 1]
```diff
@@ -120,6 +120,16 @@
   flex-direction: column;
 }
 
+.center-column-flexbox {
+  display: flex;
+  align-items: center;
+  flex-direction: column;
+}
+
+#info-error {
```
Comment (SKOLZ) [inline app/assets/stylesheets/landing.scss]: don't target id's in css

File: app/assets/stylesheets/landing.scss [hunk 2]
```diff
@@ -120,6 +120,16 @@
...
+  color: red;
```
Comment (SKOLZ) [inline app/assets/stylesheets/landing.scss]: $red

File: app/assets/stylesheets/variables/colors.scss [hunk 3]
```diff
@@ -4,3 +4,4 @@ $orange: #f5c85f;
 $gray-bg: #EEE;
 $dark-gray: #111;
 $gray: #999;
+$red: #DF0101;
```
Comment (SKOLZ) [inline app/assets/stylesheets/variables/colors.scss]: empty line at the end of the file

File: app/views/api/v1/book_suggestions/new.slim [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+      = bs.text_field :link, required: true
+    .data-seccion
+      = bs.label :publisher
+      = bs.text_field :publisher
+    .data-seccion
+      = bs.label :year
+      = bs.text_field :year
+    button type="submit"
+      | Create
+  #sectio-of-book-suggetions
```
Comment (SKOLZ) [inline app/views/api/v1/book_suggestions/new.slim]: sectio? typooo. also remove the "of"

File: app/views/api/v1/book_suggestions/new.slim [hunk 5]
```diff
@@ -0,0 +1,32 @@
+.barra-superior.background-style
```
Comment (SKOLZ) [inline app/views/api/v1/book_suggestions/new.slim]: you can do this using a border top right? I need to see the design to check this.

File: app/views/api/v1/book_suggestions/new.slim [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+.main-section.center-column-flexbox
+  = form_for @bookSuggestion, :url => "book_suggestions/create", method: :post, remote: true do |bs|
+    .data-seccion
```
Comment (SKOLZ) [inline app/views/api/v1/book_suggestions/new.slim]: don't spanglish things, it's "section"

File: app/views/api/v1/book_suggestions/new.slim [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+      = bs.label :editorial
+      = bs.text_field :editorial
+    .data-seccion
```
Comment (SKOLZ) [inline app/views/api/v1/book_suggestions/new.slim]: section

File: app/views/api/v1/book_suggestions/new.slim [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+      = bs.label :price
```
Comment (SKOLZ) [inline app/views/api/v1/book_suggestions/new.slim]: section

File: app/views/api/v1/book_suggestions/new.slim [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+      = bs.text_field :price
```
Comment (SKOLZ) [inline app/views/api/v1/book_suggestions/new.slim]: section

File: app/views/api/v1/book_suggestions/new.slim [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+    .data-seccion
```
Comment (SKOLZ) [inline app/views/api/v1/book_suggestions/new.slim]: section

File: app/views/api/v1/book_suggestions/new.slim [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+      = bs.label :author
+      = bs.text_field :author, required: true
+    .data-seccion
```
Comment (SKOLZ) [inline app/views/api/v1/book_suggestions/new.slim]: section

File: app/views/api/v1/book_suggestions/new.slim [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+      = bs.label :title
+      = bs.text_field :title, required: true
+    .data-seccion
```
Comment (SKOLZ) [inline app/views/api/v1/book_suggestions/new.slim]: section

File: app/views/api/v1/book_suggestions/new.slim [hunk 13]
```diff
@@ -0,0 +1,32 @@
...
+      = bs.label :link
+      = bs.text_field :link, required: true
+    .data-seccion
```
Comment (SKOLZ) [inline app/views/api/v1/book_suggestions/new.slim]: section

File: app/views/api/v1/book_suggestions/new.slim [hunk 14]
```diff
@@ -0,0 +1,32 @@
...
+      = bs.label :publisher
+      = bs.text_field :publisher
+    .data-seccion
```
Comment (SKOLZ) [inline app/views/api/v1/book_suggestions/new.slim]: section

File: app/assets/stylesheets/landing.scss (hunk 1 above)
Comment (gussiciliano) [inline app/assets/stylesheets/landing.scss]: Oks

File: app/assets/stylesheets/landing.scss (hunk 2 above)
Comment (gussiciliano) [inline app/assets/stylesheets/landing.scss]: Oks

File: app/assets/stylesheets/variables/colors.scss (hunk 3 above)
Comment (gussiciliano) [inline app/assets/stylesheets/variables/colors.scss]: Oks

File: app/views/api/v1/book_suggestions/new.slim (hunk 4 above)
Comment (gussiciliano) [inline app/views/api/v1/book_suggestions/new.slim]: Oks

File: app/views/api/v1/book_suggestions/new.slim (hunk 14 above)
Comment (gussiciliano) [inline app/views/api/v1/book_suggestions/new.slim]: Oks

File: app/views/api/v1/book_suggestions/new.slim (hunk 13 above)
Comment (gussiciliano) [inline app/views/api/v1/book_suggestions/new.slim]: Oks

File: app/views/api/v1/book_suggestions/new.slim (hunk 12 above)
Comment (gussiciliano) [inline app/views/api/v1/book_suggestions/new.slim]: Oks

File: app/views/api/v1/book_suggestions/new.slim (hunk 11 above)
Comment (gussiciliano) [inline app/views/api/v1/book_suggestions/new.slim]: Oks

File: app/views/api/v1/book_suggestions/new.slim (hunk 10 above)
Comment (gussiciliano) [inline app/views/api/v1/book_suggestions/new.slim]: Oks

File: app/views/api/v1/book_suggestions/new.slim (hunk 9 above)
Comment (gussiciliano) [inline app/views/api/v1/book_suggestions/new.slim]: Oks

File: app/views/api/v1/book_suggestions/new.slim (hunk 8 above)
Comment (gussiciliano) [inline app/views/api/v1/book_suggestions/new.slim]: Oks

File: app/views/api/v1/book_suggestions/new.slim (hunk 7 above)
Comment (gussiciliano) [inline app/views/api/v1/book_suggestions/new.slim]: Oks

File: app/views/api/v1/book_suggestions/new.slim (hunk 6 above)
Comment (gussiciliano) [inline app/views/api/v1/book_suggestions/new.slim]: Oks

File: app/views/api/v1/book_suggestions/new.slim (hunk 5 above)
Comment (gussiciliano) [inline app/views/api/v1/book_suggestions/new.slim]: that's right, isn't necessary this div

File: app/assets/javascripts/book_suggestions.js [hunk 15]
```diff
@@ -0,0 +1,21 @@
...
+    $.ajax({
+      method: method,
+      url: action,
+      data: data,
+      dataType: 'json',
+      success: function(data) {
+        $("#list-of-book-suggetions").append('<li>'+ data.author + ', ' + data.title + ', ' + data.link +'</li>');     
+      },
+       error: function(data) {
+        $("#info-error").html("Error!");
```
Comment (SKOLZ) [inline app/assets/javascripts/book_suggestions.js]: you changed this to a class right?

File: app/assets/javascripts/book_suggestions.js [hunk 16]
```diff
@@ -0,0 +1,21 @@
...
+    var action = $(this).attr('action');
+    var method = $(this).attr('method');
+    var data = $(this).serializeArray();
+    $.ajax({
+      method: method,
+      url: action,
+      data: data,
+      dataType: 'json',
+      success: function(data) {
+        $("#list-of-book-suggetions").append('<li>'+ data.author + ', ' + data.title + ', ' + data.link +'</li>');
```
Comment (SKOLZ) [inline app/assets/javascripts/book_suggestions.js]: you changed this to a class right?

File: app/views/api/v1/book_suggestions/new.slim [hunk 17]
```diff
@@ -1,30 +1,6 @@
...
-          $("#info-error").html(data.errors);
-        }
-      });
-    });
-  });
-
 .barra-superior.background-style
 .main-section.center-column-flexbox
-  = form_for @bookSuggestion, :url => "create", method: :post, remote: true do |bs|
+  = form_for @bookSuggestion, :url => "book_suggestions/create", method: :post, remote: true do |bs|
```
Comment (SKOLZ) [inline app/views/api/v1/book_suggestions/new.slim]: the url should be written like this:
```
url: 'INSERT_PATH_HERE'
```
where `INSERT_PATH_HERE` is the book suggestions create path you'll find when running `rake routes`

File: app/assets/javascripts/book_suggestions.js (hunk 16 above)
Comment (gussiciliano) [inline app/assets/javascripts/book_suggestions.js]: that's right, sorry

File: app/assets/javascripts/book_suggestions.js (hunk 15 above)
Comment (gussiciliano) [inline app/assets/javascripts/book_suggestions.js]: Oks

File: config/routes.rb [hunk 18]
```diff
@@ -3,9 +3,12 @@
   ActiveAdmin.routes(self)
   devise_for :users, controllers: { sessions: 'users/sessions', registrations: 'users/registrations' }
   # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
-  #get '/landing/', to: 'landing#index'
   root to: "landing#index"
 
+  get '/book_suggestions/', to: 'api/v1/book_suggestions#new'
```
Comment (Dinuuu) [inline config/routes.rb]: why don't you use resources `:book_suggestions, only: [:new, create]` ?

File: config/routes.rb [hunk 19]
```diff
@@ -3,9 +3,12 @@
...
+  get '/book_suggestions/new', to: 'api/v1/book_suggestions#new'
```
Comment (Dinuuu) [inline config/routes.rb]: Why there are 2 routes to the same method ?

File: config/routes.rb (hunk 18 above)
Comment (Dinuuu) [inline config/routes.rb]: The new method shouldn't be in the API controller since it renders a HTML, not a json

File: config/routes.rb (hunk 19 above)
Comment (gussiciliano) [inline config/routes.rb]: Oks

File: config/routes.rb (hunk 18 above)
Comment (gussiciliano) [inline config/routes.rb]: Oks

File: config/routes.rb [hunk 20]
```diff
@@ -3,9 +3,10 @@
   ActiveAdmin.routes(self)
   devise_for :users, controllers: { sessions: 'users/sessions', registrations: 'users/registrations' }
   # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
-  #get '/landing/', to: 'landing#index'
   root to: "landing#index"
 
+  get '/book_suggestions/new', to: 'book_suggestions#new'
```
Comment (Dinuuu) [inline config/routes.rb]: instead of this I would do `resources :book_suggestions, only: [:new]`

**Pull request merged**